packages/block-library/src/spacer/edit.js

```diff
@@ -197,7 +254,51 @@ const SpacerEdit = ( {
 				width: '72px',
 			} );
 		}
-	}, [] );
+		if (
+			isFlexLayout &&
+			selfStretch !== 'fill' &&
+			selfStretch !== 'fit' &&
+			! flexSize
+		) {
+			const newSize =
+				inheritedOrientation === 'horizontal'
+					? getSpacingPresetCssVar( width ) || '72px'
+					: getSpacingPresetCssVar( height ) || '100px';
+			setAttributes( {
+				style: {
+					...blockStyle,
+					layout: {
+						...layout,
+						flexSize: newSize,
+						selfStretch: 'fixed',
+					},
+				},
+			} );
+		} else if (
+			isFlexLayout &&
+			( selfStretch === 'fill' || selfStretch === 'fit' )
+		) {
+			if ( inheritedOrientation === 'horizontal' ) {
+				setAttributes( {
+					width: '0px',
+				} );
+			} else {
+				setAttributes( {
+					height: '0px',
+				} );
+			}
+		}
+	}, [
+		blockStyle,
+		flexSize,
+		height,
+		inheritedOrientation,
+		isFlexLayout,
+		layout,
+		selfStretch,
+		setAttributes,
+		width,
+	] );
```

Comment:
Does this mean that the logic here will be called more frequently now? And if so, is that a safe change? From a quick skim it sounds like it's intentional for it to be called more frequently, but I wasn't sure if there's a potential state it could get in where `setAttributes` could result in the effect being run too often 🤔

Reply:
Yeah that's a good point, I'm not 100% sure there won't be nefarious side effects. I mainly added the dependencies in because the linter was complaining about it 😅 and it didn't seem to do any harm, but negative perf impacts often don't show immediately.

Reply:
Update: I tried removing them and ended up re-adding as the effect wasn't running when a Spacer changed to "fit" or "fill" and the old width/height value was still showing up when dragging outside of the flex container. It doesn't seem to hugely increase the times the effect runs though I might easily be missing some edge case situations.